docs/src/main/sphinx/optimizer/pushdown.rst

```diff
@@ -271,3 +271,108 @@ FETCH FIRST N ROWS``.
 
 Implementation and support is connector-specific since different data sources
 support different SQL syntax and processing.
+
+We will be showcased two similar queries for demonstrating how the Trino
+compute engine deals with a connector supporting Top-N functionality and also with a
+connector which is missing this functionality.
+
+Following is  presented a concrete example of a Top-N query
+on the ``tpch`` connector which does not support Top-N pushdown functionality::
+
+    SELECT custkey, name
+    FROM tpch.sf1.customer
+    ORDER BY custkey
+    LIMIT 5;
+
+You can get the explain plan by prepending the above query with ``EXPLAIN``::
+
+    EXPLAIN SELECT custkey, name
+    FROM tpch.sf1.customer
+    ORDER BY custkey
+    LIMIT 5;
+
+Following is presented the query plan of the ``EXPLAIN`` query.
+
+.. code-block:: text
+
+ Fragment 0 [SINGLE]
+     Output layout: [custkey, name]
+     Output partitioning: SINGLE []
+     Stage Execution Strategy: UNGROUPED_EXECUTION
+     Output[custkey, name]
+     │   Layout: [custkey:bigint, name:varchar(25)]
+     │   Estimates: {rows: ? (?), cpu: ?, memory: ?, network: ?}
+     └─ TopN[5 by (custkey ASC NULLS LAST)]
+        │   Layout: [custkey:bigint, name:varchar(25)]
+        └─ LocalExchange[SINGLE] ()
+           │   Layout: [custkey:bigint, name:varchar(25)]
+           │   Estimates: {rows: ? (?), cpu: ?, memory: ?, network: ?}
+           └─ RemoteSource[1]
+                  Layout: [custkey:bigint, name:varchar(25)]
+
+ Fragment 1 [SOURCE]
+     Output layout: [custkey, name]
+     Output partitioning: SINGLE []
+     Stage Execution Strategy: UNGROUPED_EXECUTION
+     TopNPartial[5 by (custkey ASC NULLS LAST)]
+     │   Layout: [custkey:bigint, name:varchar(25)]
+     └─ TableScan[tpch:customer:sf1.0, grouped = false]
+            Layout: [custkey:bigint, name:varchar(25)]
+            Estimates: {rows: 150000 (4.58MB), cpu: 4.58M, memory: 0B, network: 0B}
+            custkey := tpch:custkey
+            name := tpch:name
+
+Note in the query above the Top-N operator ``TopN[5 by (custkey ASC NULLS LAST)]``
+is being applied in the ``SINGLE`` fragment at the compute engine level and not at the source
```

Comment:
the SINGLE/SOURCE is the partitioning of the fragment - not an identifier for the fragment.

So change to `applied in the Fragment 0`